lib/faraday/multipart/middleware.rb

```diff
@@ -28,15 +27,46 @@ def call(env)
         @app.call env
       end
 
+      private
+
+      # @param env [Faraday::Env]
+      # @yield [request_body] Body of the request
+      def match_content_type(env)
+        return unless process_request?(env)
+
+        env.request_headers[CONTENT_TYPE] ||= mime_type
+        return if env.body.respond_to?(:to_str) || env.body.respond_to?(:read)
+
+        yield(env.body)
+      end
+
+      # @param env [Faraday::Env]
+      #
+      # @return [Boolean] True if the request has a body and its Content-Type is
+      #                   urlencoded.
+      def process_request?(env)
+        type = request_type(env)
+        env.body && (type.empty? || (type == mime_type))
+      end
+
```

Comment:
And because we changed the mime_type from a class variable to an instance variable we also had to do so across this object, included in places where we once inherited from UrlEncoded. So this code is copied from that object but with the mime_type now being referenced from an instance variable instead.